### PR TITLE
Updated supported Python and dependency versions.

### PR DESCRIPTION
**Pull request by carltongibson**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ repos:
       - id: pyupgrade
```
Comment (adamchainz) [inline .pre-commit-config.yaml:5]: run `pre-commit autoupdate` to update all hooks, not just black?

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -1,24 +0,0 @@
-name: pre-commit
```
Comment (adamchainz) [inline .github/workflows/pre-commit.yml:1]: you're installing pre-commit.ci? :)

Comment (adamchainz) [reply .github/workflows/pre-commit.yml:1]: yyeeaaahhh

Comment (carltongibson) [reply .github/workflows/pre-commit.yml:1]: Hey @adamchainz — yes, indeed. 😜

File: .pre-commit-config.yaml (hunk 1 above)
Comment (carltongibson) [reply .pre-commit-config.yaml:5]: 👍 Thanks.

**Review by adamchainz — APPROVED** [review]

**Comment by adamchainz** [comment]
🚢 it

**Pull request merged**